core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -352,7 +354,7 @@ private boolean checkVersionConflict(
             } else {
                 // fatal version conflict
                 throw new VersionConflictEngineException(shardId, op.type(), op.id(),
```

Comment:
and if we do so, let's remove VersionConflictEngineException from the out catch clause?